### PR TITLE
Add more options to action settings

### PR DESCRIPTION
**Pull request by akash-codemonk**

### CODE REVIEW

File: app/client/src/pages/Editor/ConfirmRunModal.tsx [hunk 1]
```diff
@@ -19,6 +19,8 @@ class ConfirmRunModal extends React.Component<Props> {
     const { dispatch, isModalOpen } = this.props;
     const handleClose = () => {
       dispatch(showRunActionConfirmModal(false));
+
```
Comment (hetunandu) [inline app/client/src/pages/Editor/ConfirmRunModal.tsx:22]: Why are two actions needed to close this modal?

File: app/client/src/sagas/ActionExecutionSagas.ts [hunk 2]
```diff
@@ -284,8 +284,14 @@ export function* executeActionSaga(
 ) {
   const { actionId, onSuccess, onError, params } = apiAction;
   try {
-    yield put(executeApiActionRequest({ id: apiAction.actionId }));
     const api: RestAction = yield select(getAction, actionId);
+    if (api.requestConfirmation) {
+      const confirmed = yield call(confirmRunActionSaga);
+      if (!confirmed) {
+        return;
```
Comment (hetunandu) [inline app/client/src/sagas/ActionExecutionSagas.ts:291]: You might have to dispatch a failure action here so that loading states can go back to false

**Review by hetunandu — CHANGES_REQUESTED** [review]

**Comment by hetunandu** [comment]
Needs tests

File: app/client/src/pages/Editor/ConfirmRunModal.tsx (hunk 1 above)
Comment (akash-codemonk) [reply app/client/src/pages/Editor/ConfirmRunModal.tsx:22]: `showRunActionConfirmModal` is for setting the `isOpen` state of the dialog. `acceptRunActionConfirmModal ` is called when `Confirm` is clicked by the user. `cancelRunActionConfirmModal ` is called when the user clicks on `Cancel` or when the user clicks outside the dialog which will close the modal.

Here while closing I am calling  `cancelRunActionConfirmModal` to handle the scenario when user clicks outside.

File: app/client/src/sagas/ActionExecutionSagas.ts (hunk 2 above)
Comment (akash-codemonk) [reply app/client/src/sagas/ActionExecutionSagas.ts:291]: Loading is being set by `yield put(executeApiActionRequest({ id: apiAction.actionId }));` which is called after the  `requestConfirmation` check only.

**Review by hetunandu — APPROVED** [review]

**Pull request merged**